### PR TITLE
DOC: Update indexing implementation explanations.

### PR DESCRIPTION
**Pull request by seberg**
I am aware that the old stuff is a bit more in depth. But it is now hopelessly outdated, so I tried to give at least an overview of how things work now. I don't know if this actually gives an idea of what is going on, I wouldn't notice ;)...

### CODE REVIEW

**Comment by mhvk** [comment]
@seberg - I noticed this PR as the adding of the label made it float to the top. I think it is very useful, but note that in the `ufuc` section `__numpy_ufunc__` should now be mentioned, not just the logic based on `__array_priority__`. Though that does pose the problem that its behaviour is still not completely solidified; see #4815...

**Comment by seberg** [comment]
Yeah, should get back to this some time... though I don't understand what this has to do with numpy ufunc...

**Comment by mhvk** [comment]
I just read the whole section and noticed that https://github.com/numpy/numpy/pull/4331/files#diff-c4ce797b66d44da00848ce950e286105R350 did not mention `__numpy_ufunc__`, though the C code does make decisions based on its presence. But on reflection, I see that you had not touched the part on ufuncs at all here, so indeed it is not quite relevant. Sorry.

File: doc/source/reference/internals.code-explanations.rst [hunk 1]
```diff
@@ -188,139 +188,84 @@ written to so that record-array field setting works more naturally
...
-used for setting have a shape that is "broadcastable" to the shape
-implied by the indexing object.
+All python indexing operations ``arr[index]`` are organized by first preparing
+the index and finding the index type. The supported index types are:
+* integer
+* newaxis
+* slice
+* ellipsis
+* integer arrays/array-likes (fancy)
+* boolean (single boolean array); if there is more then one boolean array as
```
Comment (Nodd) [inline doc/source/reference/internals.code-explanations.rst]: Typo: "then" -> "than"

**Comment by charris** [comment]
@seberg Do you have any time to work on this for 1.10?

**Comment by seberg** [comment]
Not really, but it is probably important enough anyway ;).

**Comment by charris** [comment]
I did some basic fixes in #5791, so closing this. @seberg Could you take a look and see if anything more need clarification?

**Pull request closed without merging**